src/scripts/public/accessibility.js

```diff
@@ -20,6 +20,29 @@ const bcgovBlockThemeAccessibility = () => {
             } );
         }
 
+        // Check if the post-content id exists within the document.
+        const postContentExists = qs( '#post-content' );
+
+        // If it doesn't exist, add the id 'post-content' to the mainContent element
+        if ( ! postContentExists ) {
+            // find the most likely place for the content block to begin, starting with the wordpress default of <main>, and falling back.
+            let mainContent = null;
+            const bcgovBodyContentMain = qs( 'main' );
+            const firstH1 = qs( 'h1' );
+
+            // Set main content to the first of either main, section, or the element with a .bcgov-body-content class, or the first H1 element.
+            if ( bcgovBodyContentMain ) {
+                mainContent = bcgovBodyContentMain;
+            } else if ( firstH1 ) {
+                mainContent = firstH1;
+            }
+
+            // if the content block was found, set its ID to 'post-content' as a target for the 'skip to main content' link.
+            if ( mainContent ) {
+                mainContent.id = 'post-content';
```

Comment:
when you query either a main element or h1, although there should only be one h1, there could be multiple, both main and/or h1, if you add an id to multiple elements, then you created another issue.

Reply:
This returns the first matching element only, and querySelector() looks at the dom in presentation order, or depth-first traversal.

That guarantees that the first H1 it sees will be the first H1 in the DOM, starting from the top

reference: 
![image](https://github.com/bcgov/bcgov-wordpress-block-theme/assets/122112933/a10efd88-1a18-46d2-b142-6fffd302b5f3)

Reply:
There can only be one <main> or one <H1> result from using 

the `document.querySelector()` method, which is used in the `qs()` alias defined at https://github.com/bcgov/bcgov-wordpress-block-theme/blob/development/src/scripts/public/utils/index.js#L103

Reply:
ok, thanks for the information, this is why using qs isn't very helpful function, even though it might be short.